source/agora/crypto/Hash.d

```diff
@@ -120,60 +121,36 @@ public Hash hashFull (T) (scope const auto ref T record)
 /// Ditto
 public void hashPart (T) (scope const auto ref T record, scope HashDg state)
     /*pure*/ nothrow @nogc
-    if (is(T == struct))
 {
-    static if (is(typeof(T.init.computeHash(HashDg.init))))
-        record.computeHash(state);
-    else static if (__traits(compiles, () { const ubyte[] r = T.init[]; }))
-        state(record[]);
+    static if (is(T == struct))
+    {
+        static if (is(typeof(T.init.computeHash(HashDg.init))))
+            record.computeHash(state);
+        else static if (__traits(compiles, () { const ubyte[] r = T.init[]; }))
+            doHashWithSerializer(record[], state);
```

Comment:
Hum, IIRC this was to support `BitBlob`, where we don't want to serialize the length, so we might be able (even need) to remove this.

Reply:
Updated